src/library.js

```diff
@@ -1936,6 +1966,33 @@ LibraryManager.library = {
     stream.eof = false;
     stream.error = false;
   },
+#if EMTERPRETIFY_ASYNC
+  fclose__deps: ['close', 'fileno', '$EmterpreterAsync'],
+  fclose: function(stream) {
+    return EmterpreterAsync.handle(function(resume) {
+      // int fclose(FILE *stream);
+      // http://pubs.opengroup.org/onlinepubs/000095399/functions/fclose.html
+      var fd = _fileno(stream);
+
+      // Do syncfs on the mount and disregard the error, simulates fsync,
+      // and only returns until syncfs completes.
+      stream = FS.getStreamFromPtr(stream);
+      if (stream) {
+        var mount = stream.node.mount;
+        if (!mount.type.syncfs) {
+          resume(function() { _close(fd); });
+          return;
+        }
+        mount.type.syncfs(mount, false, function(err) {
+          // disregard err
```

Comment:
According to [the spec](http://pubs.opengroup.org/onlinepubs/000095399/functions/fclose.html) it's better to handle the error, if you think so I will do it.